onadata/apps/api/permissions.py

```diff
@@ -186,6 +172,53 @@ def has_object_permission(self, request, view, obj):
         return super().has_object_permission(request, view, obj)
 
 
+class EnketoSubmissionEditPermissions(ObjectPermissionsWithViewRestricted):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def has_object_permission(self, request, view, obj):
+        user = request.user
+        required_perms = [f'logger.{CAN_CHANGE_XFORM}']
+        is_granted_once = OneTimeAuthRequest.grant_access(request)
+
+        if is_granted_once is not None:
+            return is_granted_once
+
+        # Grant access if user is owner or super user
+        if user.is_superuser or user == obj.user:
+            return True
+
+        return user.has_perms(required_perms, obj)
+
+
+class EnketoSubmissionViewPermissions(ObjectPermissionsWithViewRestricted):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def has_permission(self, request, view):
+        return True
```

Comment:
OK, after discussing with @noliveleger, I understand that this is needed here but not in `EnketoSubmissionEditPermissions` because it's possible for the anonymous user to be allowed to view (but not edit) a submission in Enketo.